src/kOS.Safe/Screen/ScreenBuffer.cs

```diff
@@ -99,10 +99,13 @@ public virtual int ScrollVertical(int deltaRows)
         /// in need of a diff check.
         /// </summary>
         /// <param name="startRow">Starting with this row number</param>
-        /// <param name="numRows">for this many rows</param>
+        /// <param name="numRows">for this many rows, or up to the max row the buffer has if this number is too large</param>
         public void MarkRowsDirty(int startRow, int numRows)
         {
-            for( int i = 0; i < numRows ; ++i)
+            // Mark fewer rows than asked to if the reqeusted number would have blown past the end of the buffer:
+            int numSafeRows = (numRows + startRow <= buffer.Count) ? numRows : buffer.Count - startRow;
+            
+            for( int i = 0; i < numSafeRows ; ++i)
```

Comment:
I wasted some time trying to re-arrange the order in which things happen, to ensure the buffer is enlarged to the new size first before this gets called, but then ran into a chicken-and-egg problem (two steps both had to be done before each other).  I realized that was pointless because the only rows that need to be marked dirty are rows that previously existed anyway.  A newly added row starts off 'dirty' by default anyway.  Thus it's fine to call this before the buffer is expanded to the new size, provided I protect against blowing past the buffer, which was what caused the exception before.